x/gamm/keeper/estimate_swap.go

```diff
@@ -158,36 +154,32 @@ func (k Keeper) estimateSwapExactAmountOut(
 	tokenInMaxAmount sdk.Int,
 	tokenOut sdk.Coin,
 	swapFee sdk.Dec,
-) (tokenInAmount sdk.Int, err error) {
+) (updatedPool types.PoolI, tokenIn sdk.Coin, err error) {
 	if tokenInDenom == tokenOut.Denom {
-		return sdk.Int{}, errors.New("cannot trade same denomination in and out")
+		return pool, sdk.Coin{}, errors.New("cannot trade same denomination in and out")
 	}
 
 	poolOutBal := pool.GetTotalPoolLiquidity(ctx).AmountOf(tokenOut.Denom)
 	if tokenOut.Amount.GTE(poolOutBal) {
-		return sdk.Int{}, sdkerrors.Wrapf(types.ErrTooManyTokensOut,
+		return pool, sdk.Coin{}, sdkerrors.Wrapf(types.ErrTooManyTokensOut,
 			"can't get more tokens out than there are tokens in the pool")
 	}
 
-	tokenIn, err := pool.SwapInAmtGivenOut(ctx, sdk.Coins{tokenOut}, tokenInDenom, swapFee)
+	tokenIn, err = pool.SwapInAmtGivenOut(ctx, sdk.Coins{tokenOut}, tokenInDenom, swapFee)
 	if err != nil {
-		return sdk.Int{}, err
+		return pool, sdk.Coin{}, err
 	}
-	tokenInAmount = tokenIn.Amount
+	tokenInAmount := tokenIn.Amount
 
 	if tokenInAmount.LTE(sdk.ZeroInt()) {
-		return sdk.Int{}, sdkerrors.Wrapf(types.ErrInvalidMathApprox, "token amount is zero or negative")
+		return pool, sdk.Coin{}, sdkerrors.Wrapf(types.ErrInvalidMathApprox, "token amount is zero or negative")
 	}
 
 	if tokenInAmount.GT(tokenInMaxAmount) {
-		return sdk.Int{}, sdkerrors.Wrapf(types.ErrLimitMaxAmount, "Swap requires %s, which is greater than the amount %s", tokenIn, tokenInMaxAmount)
+		return pool, sdk.Coin{}, sdkerrors.Wrapf(types.ErrLimitMaxAmount, "Swap requires %s, which is greater than the amount %s", tokenIn, tokenInMaxAmount)
 	}
 
-	err = k.EstimateUpdatePoolForSwap(ctx, pool, tokenIn, tokenOut)
-	if err != nil {
-		return sdk.Int{}, err
-	}
-	return tokenInAmount, nil
+	return pool, tokenIn, nil
 }
 
 func (k Keeper) EstimateUpdatePoolForSwap(
```

Comment:
should we remove `EstimateUpdatePoolForSwap` func ?

Reply:
good catch, its not being used anywhere anymore, removed